src/library/vicon_estimator.cpp

```diff
@@ -322,18 +326,40 @@ void RotationalEstimator::updateEstimateUpdateErrorEstimate(
   Eigen::Vector3d orientation_estimate_priori_vector = orientation_estimate_priori.vec();
   Hdq
       << orientation_estimate_priori.w() * Eigen::Matrix<double, 3, 3>::Identity()
-          + skewMatrix(orientation_estimate_priori_vector), -orientation_estimate_priori.vec()
-      .transpose();
+           + skewMatrix(orientation_estimate_priori_vector),
+           -orientation_estimate_priori.vec().transpose();
   H << Hdq, Eigen::Matrix<double, 4, 3>::Zero();
-  // Predicting the measurement
-  Eigen::Quaterniond orientation_predicted = Eigen::Quaterniond(
+
+  // Calculating the measured error quaternion
+  Eigen::Quaterniond error_orientation = orientation_measured * orientation_estimate_priori.inverse();
+
+  // Calculating the predicted measurement dependant on the sign of the measured error quaternion
+  Eigen::Quaterniond orientation_predicted;
+  if (error_orientation.w() >= 0) {
+    // Assigning the flag indicating that this measurment is not flipped
+    estimator_results_.measurement_flip_flag_ = false;
+    // Calculating the predicted measurement
+    orientation_predicted = Eigen::Quaterniond(
+      Hdq * dorientation_estimate_priori + orientation_estimate_priori.coeffs());
+  } else {
+    // Assigning the flag indicating that this measurment is flipped
+    estimator_results_.measurement_flip_flag_ = true;
+    // Calculating the predicted measurement
+    orientation_predicted = Eigen::Quaterniond(
```

Comment:
Different by one minus sign. One of the minus signs that fixes the issue this pull request addresses.